### DISCUSSION TITLE
GenerateDepictionMatching2DStructure function enlarges aligned structure

### DISCUSSION THREAD

**Discussion by kathputu** [rdkit/rdkit#5858] [Q&A] [answered]
Hi, 

I am trying to achieve structure alignment with the function GenerateDepictionMatching2DStructure. 
I have my ideal structure orientation saved in a .mol file and read into python/rdkit. 

c1 is my core orientation that I want as template
m1 is the molecule that I want to align to c1

The result is what I want, however, atoms that were not present in the core are now enlarged (the bottom left part).

Does anyone know what might be the cause to this or how can this be resolved?

Thank you!

<img width="986" alt="Screen Shot 2022-12-10 at 6 01 12 PM" src="https://user-images.githubusercontent.com/89875931/206883009-682711d0-9993-4d1e-9975-267c091c0282.png">

**Comment by ptosco** [comment, marked as answer]
@kathputu that's because your core has a different scale compared to what RDKit expects; this may happen, for instance, if the core coordinates were generated with a different tool, such as MarvinJS. You will only need to normalize the size of your core such that it is in line with what RDKit expects:
```
from rdkit.Chem import rdDepictor

rdDepictor.NormalizeDepiction(core, canonicalize=0)
```
If speed is not a concern I would also recommend to use CoordGen for coordinate generation as it usually does a better job with constrained depictions that the native RDKit 2D layout engine; you will only need to
```
rdDepictor.SetPreferCoordGen(True)
```
before calling `GenerateDepictionMatching2DStructure()`.

**Reply by kathputu** [reply]
The methods you suggested achieve exactly what I am looking for. Thank you!!
